processing/src/main/java/org/apache/druid/frame/file/FrameFilePartialFetch.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.frame.file;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import org.apache.druid.java.util.common.ISE;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Response object for {@link FrameFileHttpResponseHandler}.
+ */
+public class FrameFilePartialFetch
+{
+  private final AtomicLong bytesRead = new AtomicLong(0L);
```

Comment:
Add a bit more description? The term "Response object" suggests that this is the return value from a function, indicating a point in time. In that case, I'd expect the values here to be `final` simple values that hold a snapshot of the atomic values maintained in the underlying class.

Given that these are atomic, I suspect that this class is not a simple "response object". Explain a bit?

Reply:
Ah. The handler mentioned in the comment is an HttpResponseHandler, a Druid HTTP client thing. It can mutate response objects various times (for example, on each chunk of a chunked reponse) before returning them when the response is done being handled. I'll spell that out in the comment.

Reply:
Let's see if I'm clear. The items are mutable by the thread doing the reading, but then immutable when returned to (passed to) the consumer. Correct?

If that is true, then the "Atomic" form isn't strictly necessary since only one thread at a time access them: the object ping-pongs back and forth between the two.

The question is whether the producer mutates these items while the consumer reads. If so, then the consumer will have a very hard time of it.

Reply:
You have it mostly right: it's mutable during response handling, and then it's transferred to the caller, who can do whatever they want with it. (The caller could treat the object as immutable, or could continue to mutate it.)

I think you're right that the atomic stuff isn't necessary. Only one thread at a time modifies this object, and it's transferred between threads in a synchronized way. I'll switch the implementations to be regular fields instead of atomics.